network/p2p/readSubscription.go

```diff
@@ -79,26 +79,17 @@ func (r *readSubscription) receiveLoop(wg *sync.WaitGroup) {
 			return
 		}
 
-		pid, err := messageSigningID(rawMsg)
-		if err != nil {
-			r.log.Err(err).Msg("failed to validate peer ID of incoming message")
-			return
+		validatorData, ok := rawMsg.ValidatorData.(validator.ValidatorData)
+		if !ok {
+			r.log.Fatal().Str("raw_msg", rawMsg.String()).Msg("validator data missing")
```

Comment:
Not a fan of the log fatal here.

 This would be triggered by an independent modification of `TopicValidator` that e.g. made it return before populating the data in some cases. We should definitely log a spectacular `Error`, but why not behave like  the above failure cases and `return`?

Reply:
We can, but personally I would consider it a bug if the validator returned `ValidationAccept` without populating the data. At the bare minimum, the validator should check that the message can be deserialized properly and that the origin id can be extracted. If it cannot do even this, then we should definitely reject the message.

Reply:
My issue is neither with checking that the validation data is here, nor with considering this a bug. We should certainly transmit the message that "something is rotten in the state of Denmark" far and wide.

 My issue is with crashing the node based on the input of one (externally controlled) message. A validator that accepts without populating the data has a bug, yes, but should its impact be to lose the entire blockchain?